config/v1/types_authentication.go

```diff
@@ -597,3 +650,25 @@ type TokenRequiredClaim struct {
 	// +required
 	RequiredValue string `json:"requiredValue"`
 }
+
+type TokenExpressionRule struct {
+	// CEL expression to evaluate against token claims.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=4096
+	Expression string `json:"expression"`
+
+	// Optional message shown when the validation fails.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// UserValidationRule provides the configuration for a single user validation rule.
+type TokenUserValidationRule struct {
+	// Expression must not exceed 4096 characters in length.
+	// Expression must not be empty.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=4096
+	Expression string `json:"expression,omitempty"`
```

Comment:
This should probably be required. The GoDoc should also be updated to be more user readable.